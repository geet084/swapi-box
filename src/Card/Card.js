import React from 'react'
import './Card.scss'

const Card = (props) => {

  const classes = `card ${props.choice}-bg`
  
  if (props.choice === 'people') {
    return (
      <div className={classes}>
        <span className="fave-icon">fave</span>
        <p>Name: {props.name}</p>
        <p>HomeWorld: {props.world}</p>
        <p>Population: {props.population}</p>
        <p>Species: {props.spec}</p>
      </div>
    )
  } else if (props.choice === 'planets') {
    return (
      <div className={classes}>
        <span className="fave-icon">fave</span>
        <p>Name: {props.name}</p>
        <p>Terrain: {props.terrain}</p>
        <p>Population: {props.population}</p>
        <p>Climate: {props.climate}</p>
        <p>Residents: {props.inhabitants}</p>
      </div>
    )
  } else {
    return (
      <div className={classes}>
        <span className="fave-icon">fave</span>
        <p>Name: {props.name}</p>
        <p>Model: {props.model}</p>
        <p>Class: {props.vehicle_class}</p>
        <p>Passengers: {props.passengers}</p>
      </div>
    )
  }

}

export default Card;