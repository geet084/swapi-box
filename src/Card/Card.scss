.card {
  background-color: #272323;
  width: 225px;
  color: white;
  display: flex;
  flex-direction: column;
  padding-left: 10px;
  justify-content: center;
  border: 2px solid #8e8e8e;
  margin: 10px 0px;
}