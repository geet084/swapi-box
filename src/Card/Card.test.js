import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'

describe('Card', () => {
  let wrapper;

  it('should properly render snapshot', () => {
    const mockChoice = 'people';
    const mockSelection = {name: 'hello'};
    wrapper = shallow(
      <Card choice={mockChoice} {...mockSelection} />
    )
    wrapper.setProps({choice: 'people'})
    expect(wrapper).toMatchSnapshot()
  })

  it('should properly render snapshot', () => {
    const mockChoice = 'planets';
    const mockSelection = { name: 'hello', residents: ['bob'] };
    wrapper = shallow(
      <Card choice={mockChoice} {...mockSelection} />
    )
    wrapper.setProps({ choice: 'planets' })
    expect(wrapper).toMatchSnapshot()
  })

  it('should properly render snapshot', () => {
    const mockChoice = 'vehicles';
    const mockSelection = { name: 'hello' };
    wrapper = shallow(
      <Card choice={mockChoice} {...mockSelection} />
    )
    wrapper.setProps({ choice: 'vehicles' })
    expect(wrapper).toMatchSnapshot()
  })

})