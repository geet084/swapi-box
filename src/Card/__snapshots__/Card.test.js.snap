// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should properly render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    choice="people"
    name="hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="fave-icon"
        >
          <i
            className="fas fa-jedi"
          />
        </span>,
        <p>
          Name: 
          hello
        </p>,
        <p>
          HomeWorld: 
        </p>,
        <p>
          Population: 
        </p>,
        <p>
          Species: 
        </p>,
      ],
      "className": "card people-bg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-jedi"
          />,
          "className": "fave-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-jedi",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name: ",
            "hello",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Name: ",
          "hello",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "HomeWorld: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "HomeWorld: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Population: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Species: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Species: ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="fave-icon"
          >
            <i
              className="fas fa-jedi"
            />
          </span>,
          <p>
            Name: 
            hello
          </p>,
          <p>
            HomeWorld: 
          </p>,
          <p>
            Population: 
          </p>,
          <p>
            Species: 
          </p>,
        ],
        "className": "card people-bg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-jedi"
            />,
            "className": "fave-icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-jedi",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              "hello",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            "hello",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "HomeWorld: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "HomeWorld: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Species: ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card should properly render snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    choice="planets"
    name="hello"
    residents={
      Array [
        "bob",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="fave-icon"
        >
          <i
            className="fas fa-jedi"
          />
        </span>,
        <p>
          Name: 
          hello
        </p>,
        <p>
          Terrain: 
        </p>,
        <p>
          Population: 
        </p>,
        <p>
          Climate: 
        </p>,
        <p>
          Residents: 
          bob
        </p>,
      ],
      "className": "card planets-bg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-jedi"
          />,
          "className": "fave-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-jedi",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name: ",
            "hello",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Name: ",
          "hello",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Terrain: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Terrain: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Population: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Climate: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Climate: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Residents: ",
            Array [
              "bob",
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Residents: ",
          "bob",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="fave-icon"
          >
            <i
              className="fas fa-jedi"
            />
          </span>,
          <p>
            Name: 
            hello
          </p>,
          <p>
            Terrain: 
          </p>,
          <p>
            Population: 
          </p>,
          <p>
            Climate: 
          </p>,
          <p>
            Residents: 
            bob
          </p>,
        ],
        "className": "card planets-bg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-jedi"
            />,
            "className": "fave-icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-jedi",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              "hello",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            "hello",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Terrain: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Terrain: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Climate: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Climate: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Residents: ",
              Array [
                "bob",
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Residents: ",
            "bob",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card should properly render snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    choice="vehicles"
    name="hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="fave-icon"
        >
          <i
            className="fas fa-jedi"
          />
        </span>,
        <p>
          Name: 
          hello
        </p>,
        <p>
          Model: 
        </p>,
        <p>
          Class: 
        </p>,
        <p>
          Passengers: 
        </p>,
      ],
      "className": "card vehicles-bg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-jedi"
          />,
          "className": "fave-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-jedi",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name: ",
            "hello",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Name: ",
          "hello",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Model: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Model: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Class: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Class: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Passengers: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Passengers: ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="fave-icon"
          >
            <i
              className="fas fa-jedi"
            />
          </span>,
          <p>
            Name: 
            hello
          </p>,
          <p>
            Model: 
          </p>,
          <p>
            Class: 
          </p>,
          <p>
            Passengers: 
          </p>,
        ],
        "className": "card vehicles-bg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-jedi"
            />,
            "className": "fave-icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-jedi",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              "hello",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            "hello",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Model: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Model: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Class: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Class: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passengers: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Passengers: ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
